templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -20,89 +20,121 @@ contents:
+                    RESOLVCONFTMPPATH="$(mktemp)"
+                    sed -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script" \
+                        /var/run/NetworkManager/resolv.conf > "${RESOLVCONFTMPPATH}"
+                    sed -i "0,/^nameserver.*/s//nameserver $NAMESERVER_IP\n\0/" "${RESOLVCONFTMPPATH}"
+                    if ! grep -q search "${RESOLVCONFTMPPATH}"; then
+                        # Make sure we have a search entry
+                        echo "search {{.DNS.Spec.BaseDomain}}" >> "${RESOLVCONFTMPPATH}"
+                    else
+                        # Make sure cluster domain is first in the search list
+                        sed -i "s/^search \(.*\)/search {{.DNS.Spec.BaseDomain}} \1/" "${RESOLVCONFTMPPATH}"
+                        # Remove duplicate cluster domain entries
+                        sed -i "s/\(search {{.DNS.Spec.BaseDomain}}.*\) {{.DNS.Spec.BaseDomain}}\( .*\|$\)/\1\2/" "${RESOLVCONFTMPPATH}"
+                    fi
+                    # Only leave the first 3 nameservers in /etc/resolv.conf
+                    sed -i ':a $!{N; ba}; s/\(^\|\n\)nameserver/\n# nameserver/4g' "${RESOLVCONFTMPPATH}"
+                    mv -f "${RESOLVCONFTMPPATH}" /etc/resolv.conf
+                    # Workaround for bz 1929160. Reload NetworkManager to force it to
+                    # re-run the lookup of the hostname now that we know we have DNS
+                    # servers configured correctly in resolv.conf.
+                    nmcli general reload dns-rc
                 fi
-                # Only leave the first 3 nameservers in /etc/resolv.conf
-                sed -i ':a $!{N; ba}; s/\(^\|\n\)nameserver/\n# nameserver/4g' /etc/resolv.tmp
-                mv -f /etc/resolv.tmp /etc/resolv.conf
-                # Workaround for bz 1929160. Reload NetworkManager to force it to
-                # re-run the lookup of the hostname now that we know we have DNS
-                # servers configured correctly in resolv.conf.
-                nmcli general reload dns-rc
             fi
-        fi
-        ;;
-        *)
-        ;;
-    esac
+            ;;
+            *)
+            ;;
+        esac
+    }
+
+    # Pull container image outside of the timeout in a background process to workaround possible
+    # image corruption [1]. As long as the overall script duration is below NM dispatcher timeout, this pull will continue
+    # in the background even after this script concludes [1].
+    # [1] https://github.com/containers/podman/issues/14003
+    if ! /usr/bin/podman image exists {{ .Images.baremetalRuntimeCfgImage }}; then
+        # Multiple image pulls may occur in parallel wasting bandwidth therefore this
+        # needs to be removed when podman image corruption issue is resolved.
+        # Send to background to allow it to run indepedently of dispatcher timeout. Image pull will continue as long as
+        # this script respects NM timeout.
```

Comment:
`Image pull will continue as long as this script respects NM timeout` is difficult to make sense of. Maybe add more detail or it needs to be rephrased after the latest change? Are you trying to say that if the script is killed because of it not respecting the NM timeout, the pull will be killed as well?
Sorry for the comment nits, but we are doing something weird so it might pay off to be super clear about it in the comments. Maybe
`As long as this script respects and ends by itself before the NM timeout, the pull will continue in the background without being killed by NM`

Reply:
> Are you trying to say that if the script is killed because of it not respecting the NM timeout, the pull will be killed as well?

Yes.
If its not clear, I value your input because this is indeed weird.